services/payments/secrets.go

```diff
@@ -106,6 +149,126 @@ func nitroAwsCfg(ctx context.Context) (aws.Config, error) {
 
 var errSecretsNotLoaded = errors.New("secrets are not yet loaded")
 
+func (s *Service) createVault(
```

Comment:
I like this idea but we need to think it through a bit more. For example, I can imagine a case where we need to make a new Solana address for a payout but still need to drain the prior address. It would be better to have drained before making a new address, but we wouldn't want to be locked out accidentally in a case where the order was wrong.

Reply:
Couldn't this be solved with the DR tool in the interim? Not having this and merging means that we'd be merging in secure code that isn't properly enforcing authorization controls. I'm not comfortable doing that even with the timeline we're on.

Reply:
We discussed this out of band and concluded that there is no change to the existing security posture of this mechanism and that the risks are in cases of operator negligence. To mitigate this risk, we've increased the checks that are done with the `verify` command to include decryption of the share to ensure that it was correctly encrypted with the operator key.